### PR TITLE
Fix #7099

### LINKED ISSUE #7099: Value of input[range] incorrectly rounded after mounting (step attribute ommited)
**Do you want to request a *feature* or report a *bug*?**
Bug
**What is the current behavior?**
React at the beginning rounds `input[type=range]` value to 0 or 1, and then looks at step parameter which is 0.1 for example. The result is incorrect value of input after component mounts. When I put `value` attribute after `step` attributes, it works fine.
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via https://jsfiddle.net or similar (template: https://jsfiddle.net/reactjs/69z2wepo/).**
Demo: https://jsfiddle.net/2Lm4gy5k/ Comment or uncomment line 13 and 14 to see the difference.

**What is the expected behavior?**
To always look at step parameter and then round the value (if required).
**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**
v15.1.0 (Opera, Chrome, Firefox). 

Yes, works in v.0.14.0 (https://jsfiddle.net/wrcnLugd/)

### PR DESCRIPTION
**Pull request by troydemonbreun**
Fix for #7099

### CODE REVIEW

File: src/renderers/dom/client/wrappers/ReactDOMInput.js [hunk 1]
```diff
@@ -64,6 +64,7 @@ var ReactDOMInput = {
       // Make sure we set .type before any other properties (setting .value
       // before .type means .value is lost in IE11 and below)
       type: undefined,
+      step: undefined,
```
Comment (zpao) [inline src/renderers/dom/client/wrappers/ReactDOMInput.js]: Can you add a comment about why we're setting `step` as well.

**Comment by troydemonbreun** [comment]
@zpao Comments updated

**Comment by zpao** [comment]
Sounds like what @spicyj was alluding to in #7099 so let's do it. Thanks!

**Comment by mikecao** [comment]
@troydemonbreun The min/max properties on the range input also cause the same problem. Please see #7170.

**Comment by troydemonbreun** [comment]
Yeah,  I will fix that also,  thanks!

**Pull request merged**